lib/export/formatters/dataElementFormatterCimpl6.js

```diff
@@ -0,0 +1,688 @@
+      return formattedTitle;
+    }
+    const formattedTitle = `${rightPad(`${formattedModifier}:`, this.maxPlacement - formattedModifier.length)}${de.identifier.name.trim()}`;
+    return formattedTitle;
+  }
+
+  formatBasedOns(de) {
+    const formattedBasedOns = [];
+    for (const basedOn of de.basedOn) {
+
+      const formattedBasedOn = this.getIdFromFqnAndTrackUsed(basedOn.fqn).name;
+      formattedBasedOns.push(formattedBasedOn);
+    }
+    return `${rightPad('Parent:', this.maxPlacement - 6)}${formattedBasedOns.join(', ')}`;
+  }
+
+  formatConcepts(de) {
+    const formattedConcepts = [];
+    for (const concept of de.concepts) {
+      const formattedConcept = `${this.formattedCodeFromCncpt(concept)}`;
+      formattedConcepts.push(formattedConcept);
+    }
+    return `${rightPad('Concept:', this.maxPlacement - 6)}${formattedConcepts.join(', ')}`;
+  }
+
+  formatDescription(de) {
+    const formattedDesc = `${rightPad('Description:', this.maxPlacement - 9)}"${de.description.trim()}"`;
+    return formattedDesc;
+  }
+
+  arrangeValuesByConstraintPaths(value, de) {
+    var constraintsByPaths = value.constraints.reduce((out, constraint, i) => {
+      if (constraint.lastModifiedBy.fqn != de.identifier.fqn) {
+        return out;
+      }
+
+      let combinedPaths = [];
+
+      if (constraint.path.length > 0) {
+
+        let pathsClone = [...constraint.path];
+
+        while (pathsClone.length > 0) {
+          const p = pathsClone.shift();
+
+          if (p.fqn) this.getIdFromFqnAndTrackUsed(p.fqn);
+
+          if (p.name == 'Units' && pathsClone.length > 1 && pathsClone[1].name == 'Coding') {
+            pathsClone.shift();
+            continue;
+          }
+
+          if (['code', 'Coding'].includes(p.name)) {
+            continue;
+          }
+
+          combinedPaths.push(p.name);
+        }
+      }
+
+      let combinedPath = (combinedPaths.length > 0) ? combinedPaths.join('.') : 'root';
+
+      if (!out[combinedPath]) out[combinedPath] = [];
+      out[combinedPath].push(constraint);
+
+      return out;
+    }, {});
+
+    if (Object.keys(constraintsByPaths).length && Object.keys(constraintsByPaths)[0] == 'CodeableConcept') {
+      constraintsByPaths = {'root':constraintsByPaths['CodeableConcept']};
+    }
+    return constraintsByPaths;
+  }
+
+  formatFields(de) {
+    const formattedFields = [];
+    const constrainedFields = [];
+    for (const f of de.fields) {
+      if (f.inheritance != 'inherited') {
+        const constraintsByPaths = this.arrangeValuesByConstraintPaths(f, de);
+        if (Object.keys(constraintsByPaths).length > 0) {
+          for (const path in constraintsByPaths) {
+            const fClone = f.clone();
+            fClone.constraints = constraintsByPaths[path];
+            let inserted_value = this.formatValue(fClone, de, true, path);
+            if (f.inheritance == 'overridden') {
+              for (var key in this.substituteHash) {
```

Comment:
Also, prefer `let key` instead of `var key`.

Reply:
Actually, for each of the files, this formatter is called so the scope is maintained since the class constructor is called every time. So, even though the variable may be global,  since the class is called and set for each file in shr_spec, that issue is avoided.

Reply:
But doesn't each file contain an entire namespace?  The true scope of a substitute is its element and sub-elements.  The scope of a file is a namespace.  It seems to me that a global that only gets reset per file (i.e., per namespace) is still the wrong scope for tracking substitution.